packages/kbn-securitysolution-hook-utils/src/use_observable/index.ts

```diff
@@ -22,31 +46,30 @@ export const useObservable = <Args extends unknown[], Result>(
   fn: (...args: Args) => Observable<Result>
 ): Task<Args, Result> => {
   const isMounted = useIsMounted();
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState<unknown | undefined>();
-  const [result, setResult] = useState<Result | undefined>();
   const subRef = useRef<Subscription | undefined>();
+  const reducer = createReducer<Result>();
```

Comment:
@machadoum Could you elaborate why the reducer cannot be a static function here? It seems to me that a new reducer will be created every time the hook is called now.

Reply:
I created a draft PR that uses the `Reducer` type that comes with React, let me know what you think: https://github.com/elastic/kibana/pull/132937

Reply:
I checked it, and it does look very similar to the code on my previous comment. We must declare the type twice on the `reducer` declaration and the `useReducer` call. It makes sense to move complexity from code to the type declaration. Let's merge your PR! 🚀